pkg/eventpersisterdwh/persister/persister.go

```diff
@@ -177,21 +225,65 @@ func (p *PersisterDWH) Check(ctx context.Context) health.Status {
 		p.logger.Error("Unhealthy due to context Done is closed", zap.Error(p.ctx.Err()))
 		return health.Unhealthy
 	default:
-		if p.group.FinishedCount() > 0 {
-			p.logger.Error("Unhealthy", zap.Int32("FinishedCount", p.group.FinishedCount()))
+		if p.group.FailedCount() > 0 {
```

Comment:
Since the persister will be run periodically, it's essential to monitor the `FailedCount `instead of the `FinishedCount `to prevent it from stopping after a health check.